### ISSUE TITLE
Window losing focus while Popover is open throws errors

### ISSUE THREAD

**Issue by RonnyLark** [tailwindlabs/headlessui#2018]
**What package within Headless UI are you using?**

`@headlessui/react`

**What version of that package are you using?**

Most recent (1.7.4)

**What browser are you using?**

Chrome

**Reproduction URL**

https://codesandbox.io/s/busy-rgb-r7boos?file=/src/App.tsx

**Describe your issue**

When using `Popover`, if the popover is open, and focus is taken away from the window, and is then returned to the window, an uncaught error is thrown.

Steps to reproduce in the sandbox above:
1. Click "Click me!".
2. A popover with the text "I am content" is opened.
3. Click away from the iframe (i.e. click the code to the left).
4. Click back on the iframe.
5. The following error is thrown:
```
Failed to execute 'contains' on 'Node': parameter 1 is not of type 'Node'.
```
This happens in non-iframe browsers as well.

Applying `focus` to the `Popover.Panel` doesn't help, as it doesn't trigger when the window loses focus.

**Comment by RobinMalfait** [comment]
Hey! Thank you for your bug report!
Much appreciated! 🙏

This should be fixed by #2019, and will be available in the next release.

You can already try it using:
- `npm install @headlessui/react@insiders`.
- `npm install @headlessui/vue@insiders`.

I also updated your CodeSandbox link with the latest insiders build: https://codesandbox.io/s/nifty-chebyshev-hzkuyr?file=/package.json

Thanks!

**Comment by RonnyLark** [comment]
Holy moly that was fast! Thanks!
